omniduct/databases/exasol.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import absolute_import
+
+from interface_meta import override
+
+from omniduct.utils.debug import logger
+
+from .base import DatabaseClient
+
+
+class ExasolClient(DatabaseClient):
+    """
+    This client connects to an Exasol service using the `pyexasol` python library.
+
+    Example Config
+    --------------
+
+    databases:
+        exasol_db:
+            protocol: exasol
+            host:
+              - 'localhost:8563'
+              - 'localhost:8564'
+            username: exasol_user
+            password: ****
+            schema: users
+    """
+
+    PROTOCOLS = ["exasol"]
+    DEFAULT_PORT = 8563
+    NAMESPACE_NAMES = ["schema", "table"]
+    NAMESPACE_QUOTECHAR = '"'
+    NAMESPACE_SEPARATOR = "."
+
+    @property
+    @override
+    def NAMESPACE_DEFAULT(self):
+        return {"schema": self.schema}
+
+    @override
+    def _init(self, schema=None, engine_opts=None):
+        self.__exasol = None
+
+        self.schema = schema
+        self.connection_fields += ("schema",)
+        self.engine_opts = engine_opts or {}
+
+    @override
+    def _connect(self):
+        import pyexasol
+
+        logger.info("Connecting to Exasol ...")
+        self.__exasol = pyexasol.connect(
+            dsn="{host}:{port}".format(host=self.host, port=self.port),
+            user=self.username,
+            password=self.password,
+            **self.engine_opts
+        )
+
+    @override
+    def _is_connected(self):
+        return self.__exasol is not None
+
+    @override
+    def _disconnect(self):
+        try:
+            self.__exasol.close()
+        except Exception:
+            pass
+        self.__exasol = None
+        self.schema = None
```

Comment:
`schema` is part of the Duct configuration, not a connection/state. It should not be set to `None` here.